CONTRIBUTING.md

```diff
@@ -30,6 +30,28 @@ works.
 
 This document will guide you through the contribution process.
 
+### Rules of thumb
+
+1. __Provide motivation for the change__  
+   Why will this change make the code better; does it fix a but, is it a new
+   feature, etc.
+2. __Don't cause _unnecessary_ code churn__  
+   As part of the projects strategy we maintain several lines of code, code 
+   churn might hinder back-porting changes to other branches. Also when you 
+   change a line, your name will come up in `git blame` and might hide the
+   previous writer of the code.
+3. __Keep your change-set self contained but at a reasonable size__  
+   30 files is a reasonable size limit, so unless you have a good reason to make
+   a bigger PR, try to break it into smaller pieces (still as self-contained
+   as possible), and cross-reference them. You can also mark some of them as
+   `blocked` pending the others.   
+4. __follow the code style rules__
+   As part of acceptance of a PR the changes must pass our linters. For C++ we
+   use Google's cpplint (with some ajustments) so following their [style-guide](https://github.com/google/styleguide)
+   should keep you in line.  
+   For JS we use this [ruleset](https://github.com/nodejs/node/blob/master/.eslintrc.yaml)
+   for ESLint plus some of our own rules [custom rules](https://github.com/nodejs/node/tree/master/tools/eslint-rules)
```

Comment:
This basically comes down to “run `make test`/`make lint`” before submitting a PR, right?